linux/shim/init.go

```diff
@@ -59,6 +61,29 @@ type initProcess struct {
 	rootfs   string
 }
 
+// prepareWorkdir creates Cwd if it does not exist.
+// rootfs needs to be mounted.
+func prepareWorkdir(rootfs, configJSONPath string) error {
+	var spec specs.Spec
+	configJSON, err := ioutil.ReadFile(configJSONPath)
+	if err != nil {
+		return err
+	}
+	if err := json.Unmarshal(configJSON, &spec); err != nil {
+		return err
+	}
+	if spec.Process == nil || spec.Process.Cwd == "" {
+		// unlikely
+		return nil
+	}
+	// Note that spec.Process.Cwd is unreliable and can contain unclean value like  "../../../../foo/bar...".
+	// Also, we use path.Clean() rather than filepath.Clean(), so as to make sure cleansing is done
+	// without the host filesystem context.
+	cleanCwd := path.Clean("/" + spec.Process.Cwd)
+	workdir := filepath.Join(rootfs, cleanCwd)
+	return os.MkdirAll(workdir, 0755)
```

Comment:
I think you will also need to look at the spec to see if user namespaces are configured for this container and `Mkdir` with proper ownership in mind or else the `Cwd` will be inaccessible to the process started.

Reply:
if anything, runc should create this because its already in a userns, pivot_root'ed and can be a simple `MkdirAll(cwd, 0755)` and no other code would need to be added.  

However, I'm not sure its runc job to create these dirs, however we do that for mount targets so maybe it would get accepted